Formula/filebeat.rb

```diff
@@ -0,0 +1,76 @@
+class Filebeat < Formula
+  desc "File harvester, used to fetch logs files and feed them into logstash"
+  homepage "https://www.elastic.co/products/beats/filebeat"
+  url "https://github.com/elastic/beats/archive/v1.2.3.tar.gz"
+  sha256 "8eea85de415898c362144ba533062651d8891241c738799e54cc9b17040c1fc9"
+
+  head "https://github.com/elastic/beats.git"
+
+  depends_on "go" => :build
+
+  def install
+    gopath = buildpath/"gopath"
+    (gopath/"src/github.com/elastic/beats").install Dir["{*,.git,.gitignore}"]
+
+    ENV["GOPATH"] = gopath
+
+    cd gopath/"src/github.com/elastic/beats/filebeat" do
+      system "make"
+      libexec.install "filebeat"
+      etc.install "etc/filebeat.yml"
+    end
+
+    (bin/"filebeat").write <<-EOS.undent
+      #!/bin/bash
+      #{libexec}/filebeat -c #{etc}/filebeat.yml "$@"
+    EOS
+  end
+
+  plist_options :manual => "filebeat"
+
+  def plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN"
+    "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>Program</key>
+        <string>#{opt_bin}/filebeat</string>
+        <key>RunAtLoad</key>
+        <true/>
+      </dict>
+    </plist>
+  EOS
+  end
+
+  test do
+    log_file = testpath/"log"
+    touch log_file
+
+    (testpath/"filebeat.yml").write <<-EOS.undent
+      filebeat:
+        prospectors:
+          -
+            paths:
+              - #{log_file}
+            scan_frequency: 0s
+      output:
+        file:
+          path: #{testpath}
+        console:
+          pretty: true
+    EOS
+
+    begin
+      fork { system bin/"filebeat", "-c", testpath/"filebeat.yml" }
+      sleep 5
+      log_file.append_lines "foo bar baz"
+      sleep 10
+      assert File.exist? testpath/"filebeat"
+    ensure
+      system "pkill", "filebeat"
```

Comment:
@UniqMartin sorry didn't saw this comment. The later override the preceding one. I discussed this in https://github.com/Homebrew/homebrew-core/pull/1592#discussion_r65188046

> Regarding the user specifying the configuration file it would work as the -c option can be passed multiple times and the last one is going to be the one actually used (as you can see in the test). But you may have a point. I did that because, when compiling, the generated binary expects the default configuration to be in $HOME/filebeat.yml. I though it would be better to have it in $HOMEBREW/etc/filebeat.yml (because it's more the "unix way"). So what's your opinion on this? Let the default behaviour alone and remove this wrapper? Or maybe try to see if I can change the default path during compilation to avoid using this wrapper? Or simply keep it this way?

@MikeMcQuaid was OK to keep things as they are (https://github.com/Homebrew/homebrew-core/pull/1592#discussion_r65188305).